### PR TITLE
Updates docker download steps to adapt to changes to admin password

### PR DESCRIPTION
**Pull request by DarshitChanpura**
### Description
Targeting OpenSearch 2.12.0 and above, the security demo configuration requires a custom password to be passed to setup the demo admin user. To adapt this change in docker distribution, a PR is in flight to modify the sample docker file (https://github.com/opensearch-project/opensearch-build/pull/4274). This PR intends to update the steps to guide user on how to pass this admin password.
 
### Issues Resolved
- Related to https://github.com/opensearch-project/security/issues/3695

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the BSD-3-Clause License.

### CODE REVIEW

**Comment by nateynateynate** [comment]
@DarshitChanpura - thanks for this. It looks alright to me, but do we have any tarball installation instructions that might benefit from this information as well? Perhaps we should file a separate issue. 

https://opensearch.org/docs/2.11/install-and-configure/install-opensearch/tar/

@hdhalter ?

**Comment by DarshitChanpura** [comment]
> @DarshitChanpura - thanks for this. It looks alright to me, but do we have any tarball installation instructions that might benefit from this information as well? Perhaps we should file a separate issue.
> 
> https://opensearch.org/docs/2.11/install-and-configure/install-opensearch/tar/
> 
> @hdhalter ?

These steps will be updated through this PR: https://github.com/opensearch-project/documentation-website/pull/6058

**Comment by DarshitChanpura** [comment]
@peterzhuamazon @prudhvigodithi @rishabh6788 Could I get some more reviews?

**Comment by DarshitChanpura** [comment]
@nateynateynate Could I get more reviews on this PR?

**Comment by DarshitChanpura** [comment]
@hdhalter @nateynateynate We are planning to release this feature in 2.12. Is there any feedback that needs to be addressed for this PR? If not should this considered as ready to be merged?

**Comment by DarshitChanpura** [comment]
@hdhalter @nateynateynate @krisfreedain Would you please review this PR?

**Comment by hdhalter** [comment]
> @hdhalter @nateynateynate @krisfreedain Would you please review this PR?

Yes, we can review. @Naarcha-AWS - can you please take a look at this to make sure the changes match the documentation updates? Thanks!

File: _includes/downloads/opensearch-docker.markdown [hunk 1]
```diff
@@ -4,7 +4,9 @@ The best way to try out OpenSearch is to use [Docker Compose](https://docs.docke
   - **macOS & Windows**: In Docker _Preferences_ > _Resources_, set RAM to at least 4 GB.
   - **Linux**: Ensure `vm.max_map_count` is set to at least 262144 as per the [documentation](/docs/opensearch/install/important-settings/).
 2. Download [docker-compose.yml](/samples/docker-compose.yml) into your desired directory
+  - **Note for OpenSearch 2.12 and Higher:**
+  For a successful security demo configuration setup, set the initial admin password using the `OPENSEARCH_INITIAL_ADMIN_PASSWORD` environment variable. Create a `.env` file in the same folder as your `docker-compose.yml` and include the variable with a strong password value. This configuration will be automatically utilized during the setup process.
```
Comment (Naarcha-AWS) [inline _includes/downloads/opensearch-docker.markdown:8]:
```suggestion
  When setting up the security demo configure, set the initial admin password using the `OPENSEARCH_INITIAL_ADMIN_PASSWORD` environment variable. Create a `.env` file in the same folder as your `docker-compose.yml` and include the variable with a strong password value. This configuration will be automatically utilized during the setup process.
```

**Review by Naarcha-AWS — CHANGES_REQUESTED** [review]

File: _includes/downloads/opensearch-docker.markdown (hunk 1 above)
Comment (DarshitChanpura) [reply _includes/downloads/opensearch-docker.markdown]: done

**Comment by DarshitChanpura** [comment]
@Naarcha-AWS would you please re-review this?

File: _includes/downloads/opensearch-docker.markdown [hunk 2]
```diff
@@ -4,7 +4,9 @@ The best way to try out OpenSearch is to use [Docker Compose](https://docs.docke
   - **macOS & Windows**: In Docker _Preferences_ > _Resources_, set RAM to at least 4 GB.
   - **Linux**: Ensure `vm.max_map_count` is set to at least 262144 as per the [documentation](/docs/opensearch/install/important-settings/).
 2. Download [docker-compose.yml](/samples/docker-compose.yml) into your desired directory
+  - **Note for OpenSearch 2.12 and Higher:**
+    When setting up the security demo configuration, set the initial admin password using the `OPENSEARCH_INITIAL_ADMIN_PASSWORD` environment variable. Create a `.env` file in the same folder as your `docker-compose.yml` and add the variable to `.env` file. Set the value to `OPENSEARCH_INITIAL_ADMIN_PASSWORD` variable to a strong admin password.  This configuration will be automatically utilized during the setup process.
```
Comment (Naarcha-AWS) [inline _includes/downloads/opensearch-docker.markdown:8]:
```suggestion
    When setting up the security demo configuration, set the initial admin password using the `OPENSEARCH_INITIAL_ADMIN_PASSWORD` environment variable. For more information, see [Setting up a demo configuration](https://opensearch.org/docs/latest//security/configuration/demo-configuration/).
```

Comment (Naarcha-AWS) [reply _includes/downloads/opensearch-docker.markdown:8]: We have documentation about how to set up the `env` files for each distribution. Lets link to that.

**Review by Naarcha-AWS — CHANGES_REQUESTED** [review]

File: _includes/downloads/opensearch-docker.markdown (hunk 2 above)
Comment (Naarcha-AWS) [reply _includes/downloads/opensearch-docker.markdown:8]: Here's the staging link for the linked page. The link in this PR won't work until we merge 2.12: https://1291fa08092a4a2dfd6d43f11d7bbcf3f22bf569.opensearch.org/docs/latest/security/configuration/demo-configuration/

Comment (DarshitChanpura) [reply _includes/downloads/opensearch-docker.markdown]: done

File: _includes/downloads/opensearch-docker.markdown [hunk 3]
```diff
@@ -4,7 +4,9 @@ The best way to try out OpenSearch is to use [Docker Compose](https://docs.docke
   - **macOS & Windows**: In Docker _Preferences_ > _Resources_, set RAM to at least 4 GB.
   - **Linux**: Ensure `vm.max_map_count` is set to at least 262144 as per the [documentation](/docs/opensearch/install/important-settings/).
 2. Download [docker-compose.yml](/samples/docker-compose.yml) into your desired directory
+  - **Note for OpenSearch 2.12 and Higher:**
```
[the remark this answers is not in the document]
Comment (DarshitChanpura) [reply _includes/downloads/opensearch-docker.markdown]: done.

**Review by Naarcha-AWS — APPROVED** [review]

**Comment by pajuric** [comment]
@nateynateynate @krisfreedain - Looks like this PR is approved and ready to be merged.

**Comment by pajuric** [comment]
@nateynateynate - is this something you are able to merege? What other approvals does this require?

**Review by peterzhuamazon — APPROVED** [review]

**Comment by altbrace** [comment]
The link to the documentation page is invalid

**Comment by DarshitChanpura** [comment]
Thanks for spotting that @altbrace. Raised a PR to fix it: https://github.com/opensearch-project/project-website/pull/2664

**Pull request merged**